packages/buidler-etherscan/src/solc/version.ts

```diff
@@ -0,0 +1,159 @@
+import { NomicLabsBuidlerPluginError } from "@nomiclabs/buidler/plugins";
+import SemverRange from "semver/classes/range";
+
+import { pluginName } from "../pluginContext";
+
+import {
+  MetadataAbsentError,
+  readSolcVersion,
+  VersionNotFoundError,
+} from "./metadata";
+
+const COMPILERS_LIST_URL =
+  "https://raw.githubusercontent.com/ethereum/solc-bin/gh-pages/bin/list.json";
+
+// Non-exhaustive interface for the official compiler list.
+export interface CompilersList {
+  releases: {
+    [version: string]: string;
+  };
+  latestRelease: string;
+}
+
+export class SolcVersionNumber {
```

Comment:
This class doesn't seem too useful to me, and it implies a lot of unnecessary conversion in the rest of the code.

I think it would be better to just use the string version, have a stand-alone `getLongVersion` function, and use `semver` to handle things like major/minor/patch if necessary.